test_rmw_implementation/test/test_client.cpp

```diff
@@ -0,0 +1,153 @@
+#else
+# define CLASSNAME(NAME, SUFFIX) NAME
+#endif
+
+class CLASSNAME (TestServiceServerIsAvailable, RMW_IMPLEMENTATION) : public ::testing::Test
+{
+protected:
+  void SetUp() override
+  {
+    init_options = rmw_get_zero_initialized_init_options();
+    rmw_ret_t ret = rmw_init_options_init(&init_options, rcutils_get_default_allocator());
+    ASSERT_EQ(RMW_RET_OK, ret) << rcutils_get_error_string().str;
+    init_options.enclave = rcutils_strdup("/", rcutils_get_default_allocator());
+    ASSERT_STREQ("/", init_options.enclave);
+    context = rmw_get_zero_initialized_context();
+    ret = rmw_init(&init_options, &context);
+    ASSERT_EQ(RMW_RET_OK, ret) << rcutils_get_error_string().str;
+    constexpr char node_name[] = "my_test_node";
+    constexpr char node_namespace[] = "/my_test_ns";
+    node = rmw_create_node(&context, node_name, node_namespace);
+    ASSERT_NE(nullptr, node) << rcutils_get_error_string().str;
+    const rosidl_service_type_support_t * ts =
+      ROSIDL_GET_SRV_TYPE_SUPPORT(test_msgs, srv, BasicTypes);
+    client = rmw_create_client(node, ts, "service_name_test", &rmw_qos_profile_default);
+    ASSERT_NE(nullptr, client) << rcutils_get_error_string().str;
+  }
+
+  void TearDown() override
+  {
+    rmw_ret_t ret = rmw_destroy_client(node, client);
+    EXPECT_EQ(RMW_RET_OK, ret) << rmw_get_error_string().str;
+    ret = rmw_destroy_node(node);
+    EXPECT_EQ(RMW_RET_OK, ret) << rmw_get_error_string().str;
+    ret = rmw_shutdown(&context);
+    EXPECT_EQ(RMW_RET_OK, ret) << rmw_get_error_string().str;
+    ret = rmw_context_fini(&context);
+    EXPECT_EQ(RMW_RET_OK, ret) << rmw_get_error_string().str;
+    ret = rmw_init_options_fini(&init_options);
+    EXPECT_EQ(RMW_RET_OK, ret) << rmw_get_error_string().str;
+  }
+
+  rmw_init_options_t init_options;
+  rmw_context_t context;
+  rmw_node_t * node;
+  rmw_client_t * client;
+};
+
+TEST_F(
+  CLASSNAME(TestServiceServerIsAvailable, RMW_IMPLEMENTATION),
+  rmw_service_server_is_available)
+{
+  bool is_available;
+  rmw_ret_t ret = rmw_service_server_is_available(nullptr, nullptr, &is_available);
+  EXPECT_EQ(ret, RMW_RET_ERROR) << rmw_get_error_string().str;
+  rmw_reset_error();
+
+  ret = rmw_service_server_is_available(node, nullptr, &is_available);
+  EXPECT_EQ(ret, RMW_RET_ERROR) << rmw_get_error_string().str;
+  rmw_reset_error();
+
+  ret = rmw_service_server_is_available(node, client, &is_available);
+  EXPECT_EQ(ret, RMW_RET_OK) << rmw_get_error_string().str;
+  EXPECT_FALSE(is_available) << rmw_get_error_string().str;
+  rmw_reset_error();
+
+  const rosidl_service_type_support_t * ts =
+    ROSIDL_GET_SRV_TYPE_SUPPORT(test_msgs, srv, BasicTypes);
+  rmw_service_t * service = rmw_create_service(
+    node, ts, "service_name_test",
+    &rmw_qos_profile_default);
+  ASSERT_NE(nullptr, client) << rcutils_get_error_string().str;
+  ret = rmw_service_server_is_available(node, client, &is_available);
```

Comment:
@ahcorde this one's flaky -- you may want to use one of the testing macros to wait for it.